Instructions for building server only are incorrect

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

docs

### Describe the bug

The list of specified projects (`-pl`) in the [building instructions](https://github.com/keycloak/keycloak/blob/490016569164086a8c2040faac51ef060bf7b277/docs/building.md?plain=1#L25) is not correct.

### Version

main

### Expected behavior

Build passes

### Actual behavior

```
[ERROR] Quarkus Extension Dependency Verification Error
[ERROR] Deployment artifact org.keycloak:keycloak-quarkus-server-deployment::jar:999.0.0-SNAPSHOT was found to be missing dependencies on the Quarkus extension artifacts marked with '-' below:
[ERROR] -     org.keycloak:keycloak-quarkus-server::jar
[ERROR] -     io.quarkus:quarkus-vertx-http-deployment::jar
[ERROR] -         io.quarkus:quarkus-mutiny-deployment::jar
[ERROR] -             io.quarkus:quarkus-smallrye-context-propagation-deployment::jar
[ERROR] -     io.quarkus:quarkus-vertx-deployment::jar
[ERROR] -         io.quarkus:quarkus-arc-deployment::jar
[ERROR] -         io.quarkus:quarkus-netty-deployment::jar
[ERROR] -     io.quarkus:quarkus-reactive-routes-deployment::jar
[ERROR] -         io.quarkus:quarkus-jackson-deployment::jar
[ERROR] -     io.quarkus:quarkus-resteasy-deployment::jar
[ERROR] -         io.quarkus:quarkus-resteasy-server-common-deployment::jar
[ERROR] -             io.quarkus:quarkus-resteasy-common-deployment::jar
[ERROR] -     io.quarkus:quarkus-resteasy-jackson-deployment::jar
[ERROR] -     io.quarkus:quarkus-hibernate-orm-deployment::jar
[ERROR] -         io.quarkus:quarkus-narayana-jta-deployment::jar
[ERROR] -         io.quarkus:quarkus-caffeine-deployment::jar
[ERROR] -     io.quarkus:quarkus-agroal-deployment::jar
[ERROR] -         io.quarkus:quarkus-datasource-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-h2-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-postgresql-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-mariadb-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-mysql-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-mssql-deployment::jar
[ERROR] -     io.quarkus:quarkus-jdbc-oracle-deployment::jar
[ERROR] -     io.quarkus:quarkus-core-deployment::jar
[ERROR] -     io.quarkus:quarkus-smallrye-health-deployment::jar
[ERROR] -         io.quarkus:quarkus-jsonp-deployment::jar
[ERROR] -     io.quarkus:quarkus-micrometer-deployment::jar
[ERROR] -     io.quarkus:quarkus-micrometer-registry-prometheus-deployment::jar
[ERROR] -     io.quarkus:quarkus-logging-json-deployment::jar
[ERROR] -     io.quarkus:quarkus-logging-gelf-deployment::jar
```

### How to Reproduce?

1. Clear your `.m2` folder.
2. Run:
```
./mvnw -pl quarkus/dist -am -DskipTests clean install
```

### Anything else?

_No response_

--- a/docs/building.md
+++ b/docs/building.md
@@ -18,11 +18,17 @@ To build Keycloak run:
 
     ./mvnw clean install
     
-This will build all modules and run the testsuite. 
+This will build all modules and run the testsuite.
+
+To build Keycloak with adapters run:
+
+    ./mvnw clean install -Pdistribution
 
 To build only the server run:
 
-    ./mvnw -pl quarkus/dist -am -DskipTests clean install
+    ./mvnw -pl quarkus/deployment,quarkus/dist -am -DskipTests clean install
+
+You can then find the ZIP distribution in `quarkus/dist/target` folder.
 
 ---
 **NOTE**
@@ -41,19 +47,15 @@ To enable it by default, add it to the `MAVEN_OPTS` environment variable:
 
     export MAVEN_OPTS="-Dmaven.build.cache.enabled=true"
 
-### Building Quarkus Distribution
-
-Please, take a look at this [documentation](../quarkus/README.md).
-
-## Starting Keycloak
+### Starting Keycloak
 
 To start Keycloak during development first build as specified above, then run:
 
-    ./mvnw -f testsuite/utils/pom.xml exec:java -Pkeycloak-server 
-
-When running testsuite, by default an account with username `admin` and password `admin` will be created within the master realm at start.
+    java -jar quarkus/server/target/lib/quarkus-run.jar start-dev
 
 To stop the server press `Ctrl + C`.
+
+For more details, follow the [`quarkus` module documentation](../quarkus/README.md).
 
 ## Working with the codebase
 

